models/baseModels/InvoiceItem/InvoiceItem.ts

```diff
@@ -595,6 +609,33 @@ export abstract class InvoiceItem extends Doc {
 
       return { for: ['not in', [itemNotFor]] };
     },
+    batch: async (doc: Doc) => {
+      const batches = await doc.fyo.db.getAll(ModelNameEnum.Batch, {
+        fields: ['name'],
+        filters: { item: doc.item as string },
+      });
+
+      return {
+        name: ['in', batches.map((b) => b.name as string)],
+      };
+    },
+    transferUnit: async (doc: Doc) => {
+      const conversionItems = await doc.fyo.db.getAll(
+        ModelNameEnum.UOMConversionItem,
+        {
+          fields: ['uom'],
+          filters: { parent: doc.item as string },
+        }
+      );
+      const allUoms = [
+        doc.unit,
+        ...conversionItems.map((i) => i.uom as string),
+      ];
+
+      return {
+        name: ['in', allUoms] as [string, string[]],
```

Comment:
`as [string, string[]`

is this assertion required ?. If yes please fix the types in the above line of codes.